File: AI-agent-frontend-vue3/src/api/report.ts
// Copyright (c) 2025 左岚. All rights reserved.
/**
 * 测试报告管理 API
 */
import request from './request'

// ==================== 类型定义 ====================

export interface TestReport {
    report_id: number
    name: string
    description?: string
    report_type: string
    test_case_id?: number
    agent_id?: number
    status: string
    created_by_id: number
    start_time?: string
    end_time?: string
    duration?: number
    total_cases: number
    passed_cases: number
    failed_cases: number
    skipped_cases: number
    blocked_cases: number
    executed_cases: number
    remaining_cases: number
    pass_rate: number
    execution_rate: number
    file_path?: string
    summary?: string
    created_at: string
    updated_at?: string
}

export interface TestReportDetail extends TestReport {
    executions?: TestExecution[]
    summary?: ReportSummary
}

export interface TestExecution {
    execution_id: number
    report_id: number
    testcase_id: number
    environment?: string
    executor?: string
    status: string
    start_time?: string
    end_time?: string
    duration?: number
    actual_result?: string
    error_message?: string
    created_at: string
    updated_at?: string
}

export interface ReportSummary {
    total_cases: number
    passed: number
    failed: number
    skipped: number
    error: number
    pass_rate: number
    total_duration: number
}

export interface TestReportCreate {
    name: string
    description?: string
    report_type?: string
    test_case_id?: number
    agent_id?: number
}

export interface TestReportUpdate {
    name?: string
    description?: string
    report_type?: string
    status?: string
    summary?: string
    content?: Record<string, any>
    issues?: any[]
}

export interface ReportGenerateRequest {
    name: string
    description?: string
    report_type: string
    testcase_ids: number[]
    environment?: string
    config?: Record<string, any>
}

export interface TestReportStatistics {
    total_reports: number
    generating_reports: number
    completed_reports: number
    failed_reports: number
    average_pass_rate: number
    total_test_cases: number
    passed_test_cases: number
    failed_test_cases: number
}

export interface PaginatedReports {
    items: TestReport[]
    total: number
    page: number
    page_size: number
    total_pages: number
}

// ==================== API 接口 ====================

/**
 * 创建测试报告
 */
export function createReportAPI(data: TestReportCreate) {
    return request<TestReport>({
        url: '/api/v1/reports',
        method: 'post',
        data
    })
}

/**
 * 获取测试报告列表（分页）
 */
export function getReportsAPI(params?: {
    page?: number
    page_size?: number
    keyword?: string
    report_type?: string
    status?: string
}) {
    return request<PaginatedReports>({
        url: '/api/v1/reports',
        method: 'get',
        params
    })
}

/**
 * 获取测试报告详情
 */
export function getReportAPI(reportId: number) {
    return request<TestReportDetail>({
        url: `/api/v1/reports/${reportId}`,
        method: 'get'
    })
}

/**
 * 更新测试报告
 */
export function updateReportAPI(reportId: number, data: TestReportUpdate) {
    return request<TestReport>({
        url: `/api/v1/reports/${reportId}`,
        method: 'put',
        data
    })
}

/**
 * 删除测试报告
 */
export function deleteReportAPI(reportId: number) {
    return request({
        url: `/api/v1/reports/${reportId}`,
        method: 'delete'
    })
}

/**
 * 生成测试报告
 */
export function generateReportAPI(data: ReportGenerateRequest) {
    return request<TestReport>({
        url: '/api/v1/reports/generate',
        method: 'post',
        data
    })
}

/**
 * 获取报告的执行记录
 */
export function getReportExecutionsAPI(reportId: number) {
    return request<TestExecution[]>({
        url: `/api/v1/reports/${reportId}/executions`,
        method: 'get'
    })
}

/**
 * 导出测试报告 (PDF/Excel)
 */
export function exportReportAPI(reportId: number, format: 'pdf' | 'excel') {
    return request({
        url: `/api/v1/reports/${reportId}/export/${format}`,
        method: 'get',
        responseType: 'blob'
    })
}

/**
 * 获取测试报告统计信息
 */
export function getReportStatisticsAPI(reportType?: string) {
    return request<TestReportStatistics>({
        url: '/api/v1/reports/statistics',
        method: 'get',
        params: reportType ? { report_type: reportType } : undefined
    })
}

/**
 * 下载报告文件
 */
export function downloadReportFile(reportId: number, format: 'pdf' | 'excel', filename?: string) {
    return exportReportAPI(reportId, format).then((blob) => {
        const url = window.URL.createObjectURL(blob as Blob)
        const link = document.createElement('a')
        link.href = url
        link.download = filename || `test_report_${reportId}.${format === 'pdf' ? 'pdf' : 'xlsx'}`
        document.body.appendChild(link)
        link.click()
        document.body.removeChild(link)
        window.URL.revokeObjectURL(url)
    })
}

File: AI-agent-backend-fastapi/app/schemas/report.py
# Copyright (c) 2025 左岚. All rights reserved.
"""测试报告Schema"""
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class TestReportBase(BaseModel):
    """测试报告基础模型"""
    name: str = Field(..., description="报告名称")
    description: Optional[str] = Field(None, description="报告描述")
    report_type: str = Field("execution", description="报告类型: execution/summary/detailed/custom")
    test_case_id: Optional[int] = Field(None, description="关联测试用例ID")
    agent_id: Optional[int] = Field(None, description="关联AI代理ID")


class TestReportCreate(TestReportBase):
    """创建测试报告"""
    pass


class TestReportUpdate(BaseModel):
    """更新测试报告"""
    name: Optional[str] = None
    description: Optional[str] = None
    report_type: Optional[str] = None
    status: Optional[str] = None
    summary: Optional[str] = None
    content: Optional[Dict[str, Any]] = None
    issues: Optional[List[Dict[str, Any]]] = None


class TestReportResponse(TestReportBase):
    """测试报告响应"""
    report_id: int
    status: str
    created_by_id: int
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    duration: Optional[float] = None
    total_cases: int
    passed_cases: int
    failed_cases: int
    skipped_cases: int
    blocked_cases: int
    executed_cases: int
    remaining_cases: int
    pass_rate: float
    execution_rate: float
    file_path: Optional[str] = None
    summary: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class TestReportDetail(TestReportResponse):
    """测试报告详情"""
    content: Optional[Dict[str, Any]] = None
    issues: Optional[List[Dict[str, Any]]] = None
    extra_data: Optional[Dict[str, Any]] = None


class TestReportStatistics(BaseModel):
    """测试报告统计"""
    total_reports: int = Field(..., description="总报告数")
    generating_reports: int = Field(..., description="生成中的报告数")
    completed_reports: int = Field(..., description="已完成的报告数")
    failed_reports: int = Field(..., description="失败的报告数")
    average_pass_rate: float = Field(..., description="平均通过率")
    total_test_cases: int = Field(..., description="总测试用例数")
    passed_test_cases: int = Field(..., description="通过的测试用例数")
    failed_test_cases: int = Field(..., description="失败的测试用例数")


class TestExecutionBase(BaseModel):
    """测试执行基础模型"""
    testcase_id: int = Field(..., description="测试用例ID")
    environment: Optional[str] = Field(None, description="执行环境")
    executor: Optional[str] = Field(None, description="执行器")


class TestExecutionCreate(TestExecutionBase):
    """创建测试执行"""
    report_id: int = Field(..., description="关联报告ID")


class TestExecutionUpdate(BaseModel):
    """更新测试执行"""
    status: Optional[str] = None
    actual_result: Optional[str] = None
    error_message: Optional[str] = None
    stack_trace: Optional[str] = None
    logs: Optional[str] = None


class TestExecutionResponse(TestExecutionBase):
    """测试执行响应"""
    execution_id: int
    report_id: int
    status: str
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    duration: Optional[float] = None
    actual_result: Optional[str] = None
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ReportGenerateRequest(BaseModel):
    """报告生成请求"""
    name: str = Field(..., description="报告名称")
    description: Optional[str] = Field(None, description="报告描述")
    report_type: str = Field("execution", description="报告类型")
    testcase_ids: List[int] = Field(..., description="测试用例ID列表")
    environment: Optional[str] = Field("default", description="执行环境")
    config: Optional[Dict[str, Any]] = Field(None, description="执行配置")


class ReportExportRequest(BaseModel):
    """报告导出请求"""
    report_id: int = Field(..., description="报告ID")
    format: str = Field("excel", description="导出格式: excel/pdf")
    include_details: bool = Field(True, description="是否包含详细信息")

